### PR TITLE
[FEATURE] Update to the latest OpenAPI spec (09 Aug 2024 / 080924)

### PR DESCRIPTION
**Pull request by davidc6**
### Changelog

- Update `apis` and `models` to use the latest OpenAPI spec (29 Aug 2024 / 080924)
- Update example code to reflect certain changes
- Update README
- Update Changelog

### Review

- `apis` and `modules` are generated (the only manual interventions are prefixing unused params that cause warnings and bulk rename to the `rolling` version)
- Main (manual) changes to review are in the `README`, `Cargo.toml` files and `examples` folders.

### Testing

Verified the changes using the examples / example code.

### CODE REVIEW

File: examples/falcon_host_details.rs [hunk 1]
```diff
@@ -51,7 +51,7 @@ async fn get_device_details(
     )
     .await?;
 
-    if response.errors.is_some() {
+    if !response.errors.is_empty() {
```
Comment (makr11st) [inline examples/falcon_host_details.rs:54]: I'm not sure about this change, isn't it the same?

File: src/apis/configuration.rs [hunk 2]
```diff
@@ -38,7 +38,7 @@ impl Default for Configuration {
     fn default() -> Self {
         Configuration {
             base_path: "https://api.crowdstrike.com".to_owned(),
-            user_agent: Some("OpenAPI-Generator/rolling/rust".to_owned()),
+            user_agent: Some("OpenAPI-Generator/2024-08-09T00:00:02Z/rust".to_owned()),
```
Comment (makr11st) [inline src/apis/configuration.rs:41]: I think this should stay as is: `rolling`

File: examples/falcon_host_details.rs (hunk 1 above)
Comment (davidc6) [reply examples/falcon_host_details.rs:54]: If you check [this change](https://github.com/CrowdStrike/rusty-falcon/pull/135/files#diff-c84b6d977020f0dafe666798ca03d44b562c2ebafa625ce899ff5a4d6350525cR16), it looks like `error` in no longer wrapped in an `Option` and is a "raw" `Vec` instead, unless I'm missing something?

Comment (makr11st) [reply examples/falcon_host_details.rs:54]: Makes sense, thanks for claryfication.

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [Unreleased]
 
+## [0.4.0] - 2024-08-27
+
+### Changed
+
+- Update Rusty Falcon to use the latest OpenAPI spec
+- Update example code to follow the latest OpenAPI spec changes
+- Fix links in the README
+
```
Comment (makr11st) [inline CHANGELOG.md:17]: As we have above the `Unreleased` section I would say we also may want to add a `Released` section for the code which has actually been released.

**Review by makr11st — APPROVED** [review]
I'm happy with this. We may want to verify some other API and add OAuth before 0.4.0 release.

**Pull request merged**